website/src/components/explorer/map.jsx

```diff
@@ -0,0 +1,223 @@
+// Contains code adapted from https://observablehq.com/@nrabinowitz/h3-index-inspector under the ISC license
+
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { Map } from "react-map-gl";
+import DeckGL from "@deck.gl/react";
+import { H3HexagonLayer } from "@deck.gl/geo-layers";
+import { PathStyleExtension } from "@deck.gl/extensions";
+import { WebMercatorViewport, FlyToInterpolator } from "@deck.gl/core";
+import { getRes0Cells, uncompactCells, cellToBoundary } from "h3-js";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+
+const INITIAL_VIEW_STATE = {
+  longitude: -74.012,
+  latitude: 40.705,
+  zoom: 2.5,
+  pitch: 0,
+  bearing: 0,
+};
+
+const MAP_STYLE = "mapbox://styles/mapbox/light-v11";
+
+export function ExplorerMap({
+  userInput = [],
+  userValidHex = false,
+  initialViewState = INITIAL_VIEW_STATE,
+  mapStyle = MAP_STYLE,
+  objectOnClick = undefined,
+  coordinateOnClick = undefined,
+}) {
+  const context = useDocusaurusContext();
+  const [currentInitialViewState, setCurrentInitialViewState] =
+    useState(initialViewState);
+  const [deckLoaded, setDeckLoaded] = useState(false);
+  const deckRef = useRef();
+  const res0Cells = useMemo(() => getRes0Cells().map((hex) => ({ hex })), []);
+  const res1Cells = useMemo(
+    () => uncompactCells(getRes0Cells(), 1).map((hex) => ({ hex })),
+    [],
+  );
+  const res2Cells = useMemo(
+    () => uncompactCells(getRes0Cells(), 2).map((hex) => ({ hex })),
+    [],
+  );
+
+  useEffect(() => {
+    if (userValidHex && deckRef.current) {
+      const { width, height } = deckRef.current.deck;
+
+      const viewport = new WebMercatorViewport({ width, height });
+      let minX = Infinity;
+      let minY = Infinity;
+      let maxX = -Infinity;
+      let maxY = -Infinity;
+
+      for (const hex of userInput) {
+        const coords = cellToBoundary(hex, true);
+        for (const coord of coords) {
+          if (coord[0] < minX) {
+            minX = coord[0];
+          } else if (coord[0] > maxX) {
+            maxX = coord[0];
+          }
+          if (coord[1] < minY) {
+            minY = coord[1];
+          } else if (coord[1] > maxY) {
+            maxY = coord[1];
```

Comment:
This approach might not work as expected for cells crossing the antimeridian I think. You might want to adjust each coord with 
```
function fixTransmeridianCoord(coord) {
  const lng = coord[0];
  coord[0] = lng < 0 ? lng + 360 : lng;
}
```